### DISCUSSION TITLE
How to set globally the prisma client?

### DISCUSSION THREAD

**Discussion by igmtdev** [honojs/hono#2527] [Q&A]
Can someone help me to set up the prisma client globally, so I can call anywhere on my routes, like this `c.get("prisma").user.findMany()`

**Comment by mapokapo** [comment]
I personally just have a `providers/` directory in my project structure which contains global long-lived objects like the database connection. You could have a file called `prisma.ts` with the following contents:

```ts
import { PrismaClient } from "@prisma/client";

/**
 * The Prisma client instance.
 */
export const prismaClient = new PrismaClient();
```

Then simply import `prismaClient` in any file that needs it.
